pkg/system/rm.go

```diff
@@ -78,3 +78,31 @@ func EnsureRemoveAll(dir string) error {
 		time.Sleep(100 * time.Millisecond)
 	}
 }
+
+// AtomicRemoveAll performs an atomic remove using `EnsureRemoveAll`
+// During removal the passed in dir will not be accessible.
+func AtomicRemoveAll(dir string) error {
+	// best effort to unmount, if this fails (on a mounted fs), the rename/rm will fail below accordingly
+	mount.Unmount(dir)
```

Comment:
Unless I'm missing something, I would suggest to use `unix.Unmount()` or `os.Unmount()` as `mount.Unmount()` is too complicated for this case -- it parses /proc/mountinfo which looks too much work to do for this case.